app/src/main/java/com/mailerdaemon/app/LoginActivity.kt

```diff
@@ -0,0 +1,195 @@
+                        progress_bar.visibility = View.GONE
+                        if (task.isSuccessful) {
+                            saveUser(Objects.requireNonNull(mAuth.currentUser))
+                        } else {
+                            Toast.makeText(applicationContext, "Invalid password", Toast.LENGTH_LONG).show()
+                        }
+                    }
+                } else login_password.error = "Password cannot be empty"
+            } else {
+                login_email.error = "Email cannot be empty"
+            }
+        }
+        signup.setOnClickListener {
+            startActivity(Intent(this, SignUpActivity::class.java))
+            finish()
+        }
+        callbackManager = CallbackManager.Factory.create()
+        LoginManager.getInstance().registerCallback(callbackManager,object : FacebookCallback<LoginResult> {
+            override fun onSuccess(loginResult: LoginResult) {
+                progress_bar.visibility = View.GONE
+                handleFacebookAccessToken(loginResult.accessToken)
+            }
+
+            override fun onCancel() {}
+            override fun onError(error: FacebookException) {
+                Toast.makeText(applicationContext, "If you don't have a account please signup.$error", Toast.LENGTH_LONG).show()
+            }
+        })
+        login_facebook.setOnClickListener {
+            progress_bar.visibility = View.VISIBLE
+            LoginManager.getInstance().logInWithReadPermissions(
+                    this@LoginActivity,
+                    listOf(email)
+            )
+        }
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken(getString(R.string.default_web_client_id))
+                .requestEmail()
+                .build()
+        val mGoogleSignInClient = GoogleSignIn.getClient(this, gso)
+        google_signin.setOnClickListener {
+            progress_bar.visibility = View.VISIBLE
+            startActivityForResult(mGoogleSignInClient.signInIntent, rcsignin)
+        }
+
+    }
+
+    private fun handleFacebookAccessToken(accessToken: AccessToken?) {
+        val credential = FacebookAuthProvider.getCredential(accessToken!!.token)
+        mAuth.signInWithCredential(credential)
+                .addOnCompleteListener(this) { task: Task<AuthResult?> ->
+                    if (task.isSuccessful) {
+                        val user = mAuth.currentUser
+                        saveUser(user)
+                    } else {
+                        Toast.makeText(this@LoginActivity, "Authentication failed." + task.exception,
+                                Toast.LENGTH_SHORT).show()
+                    }
+                }
+    }
+
+    private fun saveUser(user: FirebaseUser?) {
+        val model = UserModel()
+        model.name = user!!.displayName
+        model.userId = user.uid
+        model.rejectedPost = false
+        model.email = user.email
```

Comment:
all these data can be passed as constructor argument to `UserModel()`.

Reply:
woh kaise krte hai sir??

Reply:
val model = UserModel("Name","uid",.....)